transport/address.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2023 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package transport
+
+import (
+	"fmt"
+	"net"
+)
+
+type domainAddr struct {
+	network string
+	address string
+}
+
+func (a *domainAddr) Network() string {
+	return a.network
+}
+
+func (a *domainAddr) String() string {
+	return a.address
+}
+
+var _ net.Addr = (*domainAddr)(nil)
```

Comment:
I have one question, why do we need to define these `_` variables?

Reply:
I copied that from the standard library. It's a way to enforce that a struct implements an interface.
In this example, the code will fail to compile if `*domainAddr` is not a `net.Addr`.